expanderr.go

```diff
@@ -534,6 +578,30 @@ func logic(w io.Writer, buildctx *build.Context, posn string) error {
 			as.Lhs = append(as.Lhs, &ast.Ident{Name: "err"})
 		}
 
+		var outputStmt ast.Stmt = &ast.ReturnStmt{Results: e.results}
```

Comment:
This logic looks like a duplicate with the logic above? Can it be refactored into a helper function?